--- orig/accFldsMod.F90	2010-03-10 16:34:21.000000000 +0100
+++ patchd/accFldsMod.F90	2010-03-10 16:19:07.000000000 +0100
@@ -30,6 +30,8 @@
 ! !USES:
   use shr_kind_mod, only: r8 => shr_kind_r8
   use abortutils,   only: endrun
+  use mod_clm
+  use mod_regcm_param
 !
 ! !PUBLIC TYPES:
   implicit none
@@ -80,7 +82,9 @@
 
     ! Hourly average of 2m temperature.
 
-    dtime = get_step_size()
+!    dtime = get_step_size()
+    dtime = r2cdtime
+
     call init_accum_field(name='TREFAV', units='K', &
          desc='average over an hour of 2-m temperature', &
          accum_type='timeavg', accum_period=nint(3600._r8/dtime), &
@@ -253,8 +257,12 @@
 
     ! Determine calendar information
 
-    dtime = get_step_size()
-    nstep = get_nstep()
+!abt rcm    dtime = get_step_size()
+!           nstep = get_nstep()
+
+    dtime = r2cdtime
+    nstep = r2cnstep
+
     call get_curr_date (year, month, day, secs)
 
     ! Don't do any accumulation if nstep is zero
@@ -494,7 +502,9 @@
 
     ! Determine time step
 
-    nstep = get_nstep()
+!abt rcm    nstep = get_nstep()
+    nstep = r2cdtime
+
 
     ! Initialize 2m ref temperature max and min values
 
