--- orig/BalanceCheckMod.F90	2010-03-10 15:16:03.000000000 +0100
+++ patchd/BalanceCheckMod.F90	2010-03-10 14:44:04.521094657 +0100
@@ -372,7 +372,7 @@
        end if
     end do
     if ( found ) then
-       write(6,100)'soil balance error',nstep,index,errsoi_col(index)
+!abt commented       write(6,100)'soil balance error',nstep,index,errsoi_col(index)
        if (abs(errsoi_col(index)) > .10_r8) then
           write(6,*)'clm model is stopping'
           call endrun()
