--- orig/Biogeophysics2Mod.F90	2010-03-10 16:34:21.000000000 +0100
+++ patchd/Biogeophysics2Mod.F90	2010-03-10 16:20:37.658602212 +0100
@@ -342,6 +342,7 @@
        qflx_evap_tot(p) = qflx_evap_veg(p) + qflx_evap_soi(p)
        eflx_lh_tot(p)= hvap*qflx_evap_veg(p) + htvp(c)*qflx_evap_soi(p)
 
+
        ! Assign ground evaporation to sublimation from soil ice or to dew
        ! on snow or ground
 
