--- clm3.5/src/main/clm_varctl.F90	2007-02-28 06:46:27.000000000 +0100
+++ clm3.5_regcm/src/main/clm_varctl.F90	2010-01-22 00:10:30.000000000 +0100
@@ -51,6 +51,40 @@
   character(len=256) :: nrevsn                  ! restart data file name for branch run
   character(len=256) :: frivinp_rtm             ! RTM input data file name
   character(len=256) :: offline_atmdir          ! directory for input offline model atm data forcing files (Mass Store ok)
+
+!!!! abt rcm below
+  character(len=256) :: mksrf_fvegtyp           ! vegetation type data file name
+  character(len=256) :: mksrf_fsoitex           ! soil texture lnd grid
+  character(len=256) :: mksrf_fsoicol           ! soil color data file name
+  character(len=256) :: mksrf_flanwat           ! lake/water data file name
+  character(len=256) :: mksrf_fglacier          ! glacier data file name
+  character(len=256) :: mksrf_furban            ! urban data file name
+  character(len=256) :: mksrf_flai              ! LAI data file name
+  character(len=256) :: mksrf_offline_fnavyoro  ! land fraction and oro file name
+  character(len=256) :: mksrf_fisop             ! Isoprene data file name
+  character(len=256) :: mksrf_fbpin             ! B-pinene data file name
+  character(len=256) :: mksrf_fapin             ! A-pinene data file name
+  character(len=256) :: mksrf_fmbo              ! Methylbutenol data file name
+  character(len=256) :: mksrf_fmyrc             ! myrcene data file name
+  character(len=256) :: mksrf_fsabi             ! sabinene data file name
+  character(len=256) :: mksrf_flimo             ! limonene data file name
+  character(len=256) :: mksrf_fco               ! carbonmonoxide data file name
+  character(len=256) :: mksrf_focim             ! ocimene data file name
+  character(len=256) :: mksrf_facar             ! a-3carene data file name
+  character(len=256) :: mksrf_fomtp             ! other monoterpenes data file name
+  character(len=256) :: mksrf_ffarn             ! farnicene file name
+  character(len=256) :: mksrf_fbcar             ! b-caryophyllene data file name
+  character(len=256) :: mksrf_fosqt             ! other sesquiterpenes data file name
+  character(len=256) :: mksrf_fmeoh             ! methanol data file name
+  character(len=256) :: mksrf_facto             ! acetone data file name
+  character(len=256) :: mksrf_fmeth             ! methane data file name
+  character(len=256) :: mksrf_fno               ! no2/n2o/nh3 data file name
+  character(len=256) :: mksrf_facta             ! Acetaldehyde/ethanol data file name
+  character(len=256) :: mksrf_fform             ! formic acid acetic acid and formaldehyde data file name
+  character(len=256) :: mksrf_fmax              ! Saturation Maximum data file name 
+  character(len=256) :: filer_rest              ! restart file name used in output.F (not namelist option)
+!!!! abt rcm above
+
 !
 ! Landunit logic
 !
diff -x .svn -Naurb clm3.5/src/main/clm_varpar.F90 clm3.5_regcm/src/main/clm_varpar.F90
