--- orig/clmtype.F90	2010-03-10 15:16:03.000000000 +0100
+++ patchd/clmtype.F90	2010-03-10 14:44:04.561094525 +0100
@@ -551,6 +551,9 @@
 	real(r8), pointer :: sha_alf(:,:)   !shade canopy total absored by leaves (W/m**2)
 	real(r8), pointer :: sun_aperlai(:,:) !sun canopy total absorbed per unit LAI (W/m**2)
 	real(r8), pointer :: sha_aperlai(:,:) !shade canopy total absorbed per unit LAI (W/m**2)
+!abt rcm below
+   real(r8), pointer :: uvdrag(:)           !surface drag stress
+!abt rcm above
 end type pft_eflux_type
 
 !----------------------------------------------------
@@ -832,15 +835,43 @@
 ! pft VOC flux variables structure
 !----------------------------------------------------
 type, public :: pft_vflux_type
-   real(r8), pointer :: vocflx_tot(:)     !total VOC flux into atmosphere [ug C m-2 h-1]
-   real(r8), pointer :: vocflx(:,:)       !(nvoc) VOC flux [ug C m-2 h-1]
-   real(r8), pointer :: vocflx_1(:)       !vocflx(1) (for history output) [ug C m-2 h-1]
-   real(r8), pointer :: vocflx_2(:)       !vocflx(2) (for history output) [ug C m-2 h-1]
-   real(r8), pointer :: vocflx_3(:)       !vocflx(3) (for history output) [ug C m-2 h-1]
-   real(r8), pointer :: vocflx_4(:)       !vocflx(4) (for history output) [ug C m-2 h-1]
-   real(r8), pointer :: vocflx_5(:)       !vocflx(5) (for history output) [ug C m-2 h-1]
+   real(r8), pointer :: vocflx_tot(:)     !total VOC flux into atmosphere [ug  m-2 h-1]
+   real(r8), pointer :: vocflx(:,:)       !(nvoc) VOC flux [ug  m-2 h-1]
+   real(r8), pointer :: vocflx_1(:)       !vocflx(1) (for history output) [ug  m-2 h-1]
+   real(r8), pointer :: vocflx_2(:)       !vocflx(2) (for history output) [ug  m-2 h-1]
+   real(r8), pointer :: vocflx_3(:)       !vocflx(3) (for history output) [ug  m-2 h-1]
+   real(r8), pointer :: vocflx_4(:)       !vocflx(4) (for history output) [ug  m-2 h-1]
+   real(r8), pointer :: vocflx_5(:)       !vocflx(5) (for history output) [ug  m-2 h-1]
+   real(r8), pointer :: vocflx_6(:)       !vocflx(6) (for history output) [ug  m-2 h-1] abt added
+   real(r8), pointer :: vocflx_7(:)       !vocflx(7) (for history output) [ug  m-2 h-1] abt added
+   real(r8), pointer :: vocflx_8(:)       !vocflx(8) (for history output) [ug  m-2 h-1] abt added
+   real(r8), pointer :: vocflx_9(:)       !vocflx(6) (for history output) [ug  m-2 h-1] abt added
+   real(r8), pointer :: vocflx_10(:)      !vocflx(7) (for history output) [ug  m-2 h-1] abt added
+   real(r8), pointer :: vocflx_11(:)      !vocflx(8) (for history output) [ug  m-2 h-1] abt added
+   real(r8), pointer :: vocflx_12(:)      !vocflx(6) (for history output) [ug  m-2 h-1] abt added
+   real(r8), pointer :: vocflx_13(:)      !vocflx(7) (for history output) [ug  m-2 h-1] abt added
+   real(r8), pointer :: vocflx_14(:)      !vocflx(8) (for history output) [ug  m-2 h-1] abt added
+   real(r8), pointer :: vocflx_15(:)      !vocflx(6) (for history output) [ug  m-2 h-1] abt added
+   real(r8), pointer :: vocflx_16(:)      !vocflx(7) (for history output) [ug  m-2 h-1] abt added
+   real(r8), pointer :: vocflx_17(:)      !vocflx(8) (for history output) [ug  m-2 h-1] abt added
+   real(r8), pointer :: vocflx_18(:)      !vocflx(6) (for history output) [ug  m-2 h-1] abt added
+   real(r8), pointer :: vocflx_19(:)      !vocflx(7) (for history output) [ug  m-2 h-1] abt added
+   real(r8), pointer :: vocflx_20(:)      !vocflx(8) (for history output) [ug  m-2 h-1] abt added
 end type pft_vflux_type
 
+!---------------------------------------------------
+!  Accumulation variables for VOC calc  added abt  !
+!---------------------------------------------------
+type, public :: voc_accum_type
+   real(r8), pointer :: t_sum24(:,:)      !vegetation temperature from past 24hours (K)
+   real(r8), pointer :: t_sum240(:,:)     !vegetation temperature from past 240hours(K)
+   real(r8), pointer :: p_sum24su(:,:)    !PPFD from past 24hours (sunlit)
+   real(r8), pointer :: p_sum240su(:,:)   !PPFD from past 240hours(sunlit)
+   real(r8), pointer :: p_sum24sh(:,:)    !PPFD from past 24hours (shaded)
+   real(r8), pointer :: p_sum240sh(:,:)   !PPFD from past 240hours(shaded)
+   real(r8), pointer :: monlai(:,:)       ! LAI for current and previous month (gridcell,month)
+end type voc_accum_type
+
 !----------------------------------------------------
 ! pft dust flux variables structure
 !----------------------------------------------------
@@ -1245,6 +1276,7 @@
 type, public :: column_dflux_type	
    type(pft_dflux_type):: pdf_a		!pft-level dust flux variables averaged to the column
 end type column_dflux_type
+
 !----------------------------------------------------
 ! End definition of structures defined at the column_type level
 !----------------------------------------------------
@@ -1376,6 +1408,32 @@
    type(column_pstate_type):: cps_a   !column-level physical state variables averaged to gridcell
 end type gridcell_pstate_type
 
+!---------------------------------------------------
+!          VOC emission factor map      added abt  !
+!---------------------------------------------------
+type, public :: gridcell_ef_map_type
+   real(r8), pointer :: epsilon_iso(:)   !emission factor map of isoprene from RCMiso.nc abt
+   real(r8), pointer :: epsilon_mbo(:)   !emission factor map of methylbutenol
+   real(r8), pointer :: epsilon_bpin(:)  !emission factor map of b-pinene
+   real(r8), pointer :: epsilon_apin(:)  !emission factor map of a-pinene
+   real(r8), pointer :: epsilon_myrc(:)  !emission factor map of myrcene
+   real(r8), pointer :: epsilon_sabi(:)  !emission factor map of sabinene
+   real(r8), pointer :: epsilon_limo(:)  !emission factor map of limonene
+   real(r8), pointer :: epsilon_acar(:)  !emission factor map of a-3carene
+   real(r8), pointer :: epsilon_ocim(:)  !emission factor map of ocimene
+   real(r8), pointer :: epsilon_omtp(:)  !emission factor map of other monoterpenes
+   real(r8), pointer :: epsilon_farn(:)  !emission factor map of farnicene
+   real(r8), pointer :: epsilon_bcar(:)  !emission factor map of b-caryophyllene
+   real(r8), pointer :: epsilon_osqt(:)  !emission factor map of other sesquiterpenes
+   real(r8), pointer :: epsilon_meoh(:)  !emission factor map of methanol
+   real(r8), pointer :: epsilon_acto(:)  !emission factor map of acetone
+   real(r8), pointer :: epsilon_meth(:)  !emission factor map of methane
+   real(r8), pointer :: epsilon_no(:)    !emission factor map of no,n2o,nh3
+   real(r8), pointer :: epsilon_acta(:)  !emission factor map of acetaldehyde/ethanol
+   real(r8), pointer :: epsilon_form(:)  !emission factor map of form
+   real(r8), pointer :: epsilon_co(:)    !emission factor map of co
+end type gridcell_ef_map_type
+
 !----------------------------------------------------
 ! gridcell energy state variables structure
 !----------------------------------------------------
@@ -1652,6 +1710,9 @@
    type(pft_cstate_type) :: pc13s       !pft carbon-13 state
    type(pft_cflux_type)  :: pc13f       !pft carbon-13 flux
    
+   ! voc accumulation vars !!abt added!!
+   type(voc_accum_type)  :: pva         !accumulation variables from VOC calc
+
 end type pft_type
 
 !----------------------------------------------------
@@ -1812,6 +1873,7 @@
    type(gridcell_vflux_type) :: gvf	!average of VOC fluxes all landunits
    type(gridcell_dflux_type) :: gdf	!average of dust fluxes all landunits
 
+   type(gridcell_ef_map_type):: gem     !Emission factor map at grid level (epsilon) abt added
 end type gridcell_type
 
 !----------------------------------------------------
