--- clm3.5/src/main/histFldsMod.F90	2007-04-26 20:50:19.000000000 +0200
+++ clm3.5_regcm/src/main/histFldsMod.F90	2010-01-22 00:10:30.000000000 +0100
@@ -322,29 +322,88 @@
 #endif
 
 #if (defined VOC)
-    call add_fld1d (fname='VOCFLXT', units='uG/M2/H',  &
-         avgflag='A', long_name='total VOC flux into atmosphere', &
-         ptr_pft=clm3%g%l%c%p%pvf%vocflx_tot, set_lake=0._r8)
+!abt removed this    call add_fld1d (fname='VOCFLXT', units='uG/M2/H',  &
+!         avgflag='A', long_name='total VOC flux into atmosphere', &
+!         ptr_pft=clm3%g%l%c%p%pvf%vocflx_tot, set_lake=0._r8)
 
     call add_fld1d (fname='ISOPRENE', units='uG/M2/H',  &
          avgflag='A', long_name='isoprene flux', &
          ptr_pft=clm3%g%l%c%p%pvf%vocflx_1, set_lake=0._r8)
-
-    call add_fld1d (fname='MONOTERP', units='uG/M2/H',  &
-         avgflag='A', long_name='monoterpene flux', &
+!abt added below
+    call add_fld1d (fname='METHYLBUT', units='uG/M2/H',  &
+         avgflag='A', long_name='methylbutenol flux', &
+         ptr_pft=clm3%g%l%c%p%pvf%vocflx_6, set_lake=0._r8)
+    call add_fld1d (fname='APINE', units='uG/M2/H',  &
+         avgflag='A', long_name='alpha-pinene flux', &
+         ptr_pft=clm3%g%l%c%p%pvf%vocflx_7, set_lake=0._r8)
+    call add_fld1d (fname='BPINE', units='uG/M2/H',  &
+         avgflag='A', long_name='beta-pinene flux', &
+         ptr_pft=clm3%g%l%c%p%pvf%vocflx_8, set_lake=0._r8)
+    call add_fld1d (fname='MYRCENE', units='uG/M2/H',  &
+         avgflag='A', long_name='myrcene flux', &
          ptr_pft=clm3%g%l%c%p%pvf%vocflx_2, set_lake=0._r8)
-
-    call add_fld1d (fname='OVOC', units='uG/M2/H',  &
-         avgflag='A', long_name='other VOC flux', &
+    call add_fld1d (fname='SABINENE', units='uG/M2/H',  &
+         avgflag='A', long_name='sabinene flux', &
          ptr_pft=clm3%g%l%c%p%pvf%vocflx_3, set_lake=0._r8)
-
-    call add_fld1d (fname='ORVOC', units='uG/M2/H',  &
-         avgflag='A', long_name='other reactive VOC flux', &
+    call add_fld1d (fname='LIMONENE', units='uG/M2/H',  &
+         avgflag='A', long_name='limonene flux', &
          ptr_pft=clm3%g%l%c%p%pvf%vocflx_4, set_lake=0._r8)
-
     call add_fld1d (fname='BIOGENCO', units='uG/M2/H',  &
-         avgflag='A', long_name='biogenic CO flux', &
+         avgflag='A', long_name='biogenic co flux', &
          ptr_pft=clm3%g%l%c%p%pvf%vocflx_5, set_lake=0._r8)
+    call add_fld1d (fname='OCIMENE', units='uG/M2/H',  &
+         avgflag='A', long_name='ocimene flux', &
+         ptr_pft=clm3%g%l%c%p%pvf%vocflx_9, set_lake=0._r8)
+    call add_fld1d (fname='ACAR', units='uG/M2/H',  &
+         avgflag='A', long_name='A-3carene flux', &
+         ptr_pft=clm3%g%l%c%p%pvf%vocflx_10, set_lake=0._r8)
+    call add_fld1d (fname='OMTP', units='uG/M2/H',  &
+         avgflag='A', long_name='other monoterpenes flux', &
+         ptr_pft=clm3%g%l%c%p%pvf%vocflx_11, set_lake=0._r8)
+    call add_fld1d (fname='FARN', units='uG/M2/H',  &
+         avgflag='A', long_name='farnicene flux', &
+         ptr_pft=clm3%g%l%c%p%pvf%vocflx_12, set_lake=0._r8)
+    call add_fld1d (fname='BCAR', units='uG/M2/H',  &
+         avgflag='A', long_name='b-caryophyllene flux', &
+         ptr_pft=clm3%g%l%c%p%pvf%vocflx_13, set_lake=0._r8)
+    call add_fld1d (fname='OSQT', units='uG/M2/H',  &
+         avgflag='A', long_name='other sesquiterpenes flux', &
+         ptr_pft=clm3%g%l%c%p%pvf%vocflx_14, set_lake=0._r8)
+    call add_fld1d (fname='METHANOL', units='uG/M2/H',  &
+         avgflag='A', long_name='biogenic methanol flux', &
+         ptr_pft=clm3%g%l%c%p%pvf%vocflx_15, set_lake=0._r8)
+    call add_fld1d (fname='ACETONE', units='uG/M2/H',  &
+         avgflag='A', long_name='acetone flux', &
+         ptr_pft=clm3%g%l%c%p%pvf%vocflx_16, set_lake=0._r8)
+    call add_fld1d (fname='BIOGENMETH', units='uG/M2/H',  &
+         avgflag='A', long_name='biogenic methane flux', &
+         ptr_pft=clm3%g%l%c%p%pvf%vocflx_17, set_lake=0._r8)
+    call add_fld1d (fname='BIOGENNOS', units='uG/M2/H',  &
+         avgflag='A', long_name='NO2/N2O/NH3 flux', &
+         ptr_pft=clm3%g%l%c%p%pvf%vocflx_18, set_lake=0._r8)
+    call add_fld1d (fname='BIOGENACTA', units='uG/M2/H',  &
+         avgflag='A', long_name='biogenic ethanol/acetaldehyde flux', &
+         ptr_pft=clm3%g%l%c%p%pvf%vocflx_19, set_lake=0._r8)
+    call add_fld1d (fname='BIOGENFORM', units='uG/M2/H',  &
+         avgflag='A', long_name='biogenic formic acid/acetic acid/formaldehyde flux', &
+         ptr_pft=clm3%g%l%c%p%pvf%vocflx_20, set_lake=0._r8)
+
+!abt removed below    call add_fld1d (fname='MONOTERP', units='uG/M2/H',  &
+!         avgflag='A', long_name='monoterpene flux', &
+!         ptr_pft=clm3%g%l%c%p%pvf%vocflx_2, set_lake=0._r8)
+!
+!    call add_fld1d (fname='OVOC', units='uG/M2/H',  &
+!         avgflag='A', long_name='other VOC flux', &
+!         ptr_pft=clm3%g%l%c%p%pvf%vocflx_3, set_lake=0._r8)
+!
+!    call add_fld1d (fname='ORVOC', units='uG/M2/H',  &
+!         avgflag='A', long_name='other reactive VOC flux', &
+!         ptr_pft=clm3%g%l%c%p%pvf%vocflx_4, set_lake=0._r8)
+!
+!    call add_fld1d (fname='BIOGENCO', units='uG/M2/H',  &
+!         avgflag='A', long_name='biogenic CO flux', &
+!abt removed above         ptr_pft=clm3%g%l%c%p%pvf%vocflx_5, set_lake=0._r8)
+!!!!abt added above
 #endif
 
     ! Hydrology
diff -x .svn -Naurb clm3.5/src/main/initGridCellsMod.F90 clm3.5_regcm/src/main/initGridCellsMod.F90
