--- orig/histFileMod.F90	2010-03-10 16:34:21.000000000 +0100
+++ patchd/histFileMod.F90	2010-03-10 16:29:37.408601048 +0100
@@ -2010,6 +2010,9 @@
     use clm_varpar   , only : lsmlon, lsmlat, nlevsoi
     use ncdio
     use spmdGathScatMod, only : gather_data_to_master
+!abt added below
+    use clm_varsur     , only : glonc,glatc
+!abt added above
 !
 ! !ARGUMENTS:
     implicit none
@@ -2216,12 +2219,22 @@
           call ncd_ioglobal(varname='latrof', data=runoff%rlat, ncid=nfid(t), flag='write')
        endif
 #endif
-       call ncd_iolocal(varname='longxy'  , data=ldomain%lonc, &
-            dim1name='gridcell', ncid=nfid(t), &
-            flag='write', nlonxy=ldomain%ni, nlatxy=ldomain%nj)
-       call ncd_iolocal(varname='latixy'  , data=ldomain%latc, &
-            dim1name='gridcell', ncid=nfid(t), &
-            flag='write', nlonxy=ldomain%ni, nlatxy=ldomain%nj)
+!original commented out below by abt
+!       call ncd_iolocal(varname='longxy'  , data=ldomain%lonc, &
+!            dim1name='gridcell', ncid=nfid(t), &
+!            flag='write', nlonxy=ldomain%ni, nlatxy=ldomain%nj)
+!       call ncd_iolocal(varname='latixy'  , data=ldomain%latc, &
+!            dim1name='gridcell', ncid=nfid(t), &
+!            flag='write', nlonxy=ldomain%ni, nlatxy=ldomain%nj)
+!abt above
+!abt added below
+       if (masterproc) then
+          call ncd_ioglobal(varname='longxy', data=glonc, ncid=nfid(t), flag='write')
+       endif
+       if (masterproc) then
+          call ncd_ioglobal(varname='latixy', data=glatc, ncid=nfid(t), flag='write')
+       endif
+!abt added above
        call ncd_iolocal(varname='area'    , data=ldomain%area, &
             dim1name='gridcell', ncid=nfid(t), &
             flag='write', nlonxy=ldomain%ni, nlatxy=ldomain%nj)
@@ -3195,6 +3208,10 @@
 
         call mpi_bcast (ntapes, 1, mpi_integer, 0, mpicom, ier)
         do t = 1,ntapes
+!abt added below
+           tape(t)%nhtfrq = hist_nhtfrq(t)
+           tape(t)%mfilt  = hist_mfilt(t)
+!abt added above
            call mpi_bcast (tape(t)%nflds     , 1, MPI_INTEGER, 0, mpicom, ier)
            call mpi_bcast (tape(t)%ntimes    , 1, MPI_INTEGER, 0, mpicom, ier)
            call mpi_bcast (tape(t)%nhtfrq    , 1, MPI_INTEGER, 0 ,mpicom, ier)
@@ -3616,7 +3633,7 @@
       write(cdate,'(i4.4,"-",i2.2,"-",i2.2,"-",i5.5)') yr,mon,day,sec
    endif
    write(hist_index,'(i1.1)') hist_file - 1
-   set_hist_filename = "./"//trim(caseid)//".clm2.h"//hist_index//"."//&
+   set_hist_filename = "./output/"//trim(caseid)//".clm2.h"//hist_index//"."//&
         trim(cdate)//".nc"
 
   end function set_hist_filename
