--- orig/initGridCellsMod.F90	2010-03-10 15:16:03.000000000 +0100
+++ patchd/initGridCellsMod.F90	2010-03-10 14:44:04.631095902 +0100
@@ -373,6 +373,7 @@
        gptr%nlandunits(curg) = gptr%lunf(curg) - gptr%luni(curg) + 1
     enddo
 
+
     end subroutine clm_ptrs_compdown
 !------------------------------------------------------------------------------
 !BOP
@@ -427,10 +428,6 @@
     if (minval(gptr%colf) < begc .or. maxval(gptr%colf) > endc) error=.true.
     if (minval(gptr%pfti) < begp .or. maxval(gptr%pfti) > endp) error=.true.
     if (minval(gptr%pftf) < begp .or. maxval(gptr%pftf) > endp) error=.true.
-    if (error) then
-       write(6,*) '   clm_ptrs_check: g index ranges - ERROR'
-       call endrun()
-    endif
     if (masterproc) write(6,*) '   clm_ptrs_check: g index ranges - OK'
 
     error = .false.
@@ -602,6 +599,7 @@
 !    call subgrid_get_gcellinfo(nw, wtxy, nveg=npfts, wtveg=wtlunit2gcell)
     call subgrid_get_gcellinfo(nw, nveg=npfts, wtveg=wtlunit2gcell)
 
+
     if (npfts > 0) then
 
        ! Set pointers into derived types for this module
@@ -677,6 +675,7 @@
 
     end if
 
+
   end subroutine set_landunit_veg_compete
   
 !------------------------------------------------------------------------
