--- 1/ncdio.F90	2010-09-23 17:24:12.000000000 +0200
+++ 2//ncdio.F90	2010-09-23 17:25:05.782000945 +0200
@@ -1811,7 +1811,7 @@
     use shr_kind_mod, only : r8 => shr_kind_r8
     use decompMod   , only : get_proc_bounds
     use clm_varpar  , only : maxpatch
-    use nanMod      , only : nan
+    use nanMod      , only : bigint
 !
 ! !ARGUMENTS:
     implicit none
@@ -1928,10 +1928,10 @@
          print *, 'NETCDF ERROR: ', NF_STRERROR(ret)
        end if
 
-       cols(:)=nan
-       pfts(:)=nan
-       col_offset=nan
-       pi_offset=nan
+       cols(:)=bigint
+       pfts(:)=bigint
+       col_offset=bigint
+       pi_offset=bigint
        i=1
        do cc = 1, totcols
           if (cols1dlon(cc).eq.lonidx.and.cols1dlat(cc).eq.latidx) then
