--- orig/perf_mod.F90	2010-03-10 15:16:03.000000000 +0100
+++ patchd/perf_mod.F90	2010-03-10 14:44:04.651093881 +0100
@@ -760,18 +760,20 @@
                           timing_barrier_out=profile_barrier, &
                           perf_single_file_out=profile_single_file )
     if ( MasterTask2 ) then
-       unitn = shr_file_getUnit()
-       write(6,*) 'Read in prof_inparm namelist from: '//trim(NLFilename)
-       ierr = 1
-       open( unitn, file=trim(NLFilename), status='old', iostat=ierr )
-       if (ierr .eq. 0) then
-          ierr = 1
-          do while( ierr > 0 )
-             read(unitn, nml=prof_inparm, iostat=ierr)
-          end do
-          close(unitn)
-       endif
-       call shr_file_freeUnit( unitn )
+!       unitn = shr_file_getUnit()
+!       write(6,*) 'Read in prof_inparm namelist from: '//trim(NLFilename)
+!       ierr = 1
+!abt comment below
+!       open( unitn, file=trim(NLFilename), status='old', iostat=ierr )
+!       if (ierr .eq. 0) then
+!          ierr = 1
+!          do while( ierr > 0 )
+!             read(unitn, nml=prof_inparm, iostat=ierr)
+!          end do
+!          close(unitn)
+!       endif
+!       call shr_file_freeUnit( unitn )
+       profile_disable = .true.
     endif
     ! This logic assumes that there will be only one MasterTask
     ! per communicator, and that this MasterTask is process 0.
