--- clm3.5/src/main/restFileMod.F90	2007-04-30 22:20:27.000000000 +0200
+++ clm3.5_regcm/src/main/restFileMod.F90	2010-01-22 00:10:30.000000000 +0100
@@ -587,6 +587,7 @@
        call opnfil( filename, nio, 'f' )
        
        if (mss_irt == 0) then
+!abt changed          write(nio,'(a)') fnamer
           write(nio,'(a)') fnamer
        else
           rem_dir = trim(archive_dir) // '/rest/'
@@ -688,15 +689,15 @@
        write(cdate,'(i4.4,"-",i2.2,"-",i2.2,"-",i5.5)') yr,mon,day,sec
        
        if (trim(type) == 'binary') then
-          restFile_filename = "./"//trim(caseid)//".clm2.r."//trim(cdate)
+          restFile_filename = "./output/"//trim(caseid)//".clm2.r."//trim(cdate)
        else if (trim(type) == 'netcdf') then
-          if (present(offset)) then
-             restFile_filename = "./"//trim(caseid)//".clm2.i."//trim(cdate)//".nc"
-             write(6,*)'writing initial file ',trim(restFile_filename),' for model date = ',cdate
-          else
-             restFile_filename = "./"//trim(caseid)//".clm2.r."//trim(cdate)//".nc"
+!abt commented          if (present(offset)) then
+!             restFile_filename = "./"//trim(caseid)//".clm2.i."//trim(cdate)//".nc"
+!             write(6,*)'writing initial file ',trim(restFile_filename),' for model date = ',cdate
+!          else
+             restFile_filename = "./output/"//trim(caseid)//".clm2.r."//trim(cdate)//".nc"
              write(6,*)'writing restart file ',trim(restFile_filename),' for model date = ',cdate
-          end if
+!          end if
        else
           write(6,*)'restart file type ',trim(type),' is not supported'; call endrun()
        end if
@@ -793,7 +794,6 @@
        call check_ret( nf_def_dim(ncid, 'rtmlat'  , rtmlat         , dimid), subname )
 #endif
        call check_ret( nf_def_dim(ncid, 'string_length', 64        , dimid), subname)
-       
        ! Define global attributes
        
        str = 'CF-1.0'
diff -x .svn -Naurb clm3.5/src/main/subgridMod.F90 clm3.5_regcm/src/main/subgridMod.F90
