--- orig/shr_orb_mod.F90	2010-03-10 15:16:03.000000000 +0100
+++ patchd/shr_orb_mod.F90	2010-03-10 14:44:04.651093881 +0100
@@ -8,7 +8,9 @@
    use shr_kind_mod
    use shr_sys_mod
    use shr_const_mod
-
+!abt added below
+   use clm_varsur, only: numdays
+!abt added above
    IMPLICIT none
 
    !----------------------------------------------------------------------------
@@ -548,7 +550,8 @@
    real   (SHR_KIND_R8),intent(out) :: eccf   ! Earth-sun distance factor (ie. (1/r)**2)
  
    !---------------------------Local variables-----------------------------
-   real   (SHR_KIND_R8),parameter :: dayspy = 365.0_SHR_KIND_R8  ! days per year
+!abt commented   real   (SHR_KIND_R8),parameter :: dayspy = 365.0_SHR_KIND_R8  ! days per year
+   real   (SHR_KIND_R8)           :: dayspy                      ! days per year
    real   (SHR_KIND_R8),parameter :: ve     = 80.5_SHR_KIND_R8   ! Calday of vernal equinox
                                                      ! assumes Jan 1 = calday 1
  
@@ -574,7 +577,9 @@
    ! an increment (delta lambda m in Berger 1978) that is the number of
    ! days past or before (a negative increment) the vernal equinox divided by
    ! the days in a model year times the 2*pi radians in a complete orbit.
- 
+!abt added below
+   dayspy = numdays
+!abt added above 
    lambm = lambm0 + (calday - ve)*2._SHR_KIND_R8*pi/dayspy
    lmm   = lambm  - mvelpp
  
