diff -Naurb orig/clm_time_manager.F90 patch/clm_time_manager.F90
--- orig/clm_time_manager.F90	2010-07-15 14:47:01.000000000 +0200
+++ patch/clm_time_manager.F90	2010-07-15 14:09:16.000000000 +0200
@@ -627,7 +627,7 @@
   ! Advance the timestep.  
   ! Data from the restart file corresponds to the last timestep of the previous run.
 
-  call advance_timestep()
+  call advance_timestep()
 
   ! Set flag that this is the first timestep of the restart run.
 
