--- orig/subgridMod.F90	2010-03-10 16:34:21.000000000 +0100
+++ patchd/subgridMod.F90	2010-03-10 16:19:07.000000000 +0100
@@ -227,7 +227,7 @@
     real(r8), optional, intent(out) :: wtglacier  ! weight (relative to gridcell) of glacier landunitof glacier pfts (columns) in glacier landunit
 !
 ! !CALLED FROM:
-! subroutines decomp_init, initGridCells
+! subroutines initGridCells
 !
 ! !REVISION HISTORY:
 ! 2002.09.11  Mariana Vertenstein  Creation.
